### ISSUE TITLE
test_vfs_works fails on Windows

### ISSUE THREAD

**Issue by vipentti** [rust-analyzer/rust-analyzer#827]
Currently `test_vfs_works` fails on Windows (at least on my machine(s)). I'm not sure if this is related to #734 .

I have not debugged this exactly, but this seems to be "fixed" by increasing `num_tasks` given to `process_tasks`.

https://github.com/rust-analyzer/rust-analyzer/blob/b85c189500a47f3c5f7690250611b06e08616c9a/crates/ra_vfs/tests/vfs.rs#L121

When on Windows, you increase the `num_tasks` to `4`, the tests pass correctly. However, this causes the tests to fail on Linux? (at least through WSL).

**NB** Looking at the logs, it seems like on Windows you get some extra events compared to Linux. Which would explain why increasing `num_tasks` fixes this issue.

<details>
    <summary>Windows output with num_tasks 2</summary>

```
running 1 test
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Create(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1\\sub2"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1\\sub2\\new.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Rename(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1\\sub2\\new.rs",
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1\\sub2\\new1.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmpffj1FC\\a\\sub1\\sub2"
    )
)
test test_vfs_works ... FAILED

failures:

---- test_vfs_works stdout ----
[crates\ra_vfs\tests\vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        0
    ),
    files: [
        (
            "baz.rs",
            "nested hello"
        )
    ]
}
[crates\ra_vfs\tests\vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        1
    ),
    files: [
        (
            "bar.rs",
            "world"
        ),
        (
            "foo.rs",
            "hello"
        )
    ]
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "quux"
    )
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "corge"
    )
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: Some(
        "new hello"
    )
}
[crates\ra_vfs\tests\vfs.rs:120] "before rename" = "before rename"
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1",
    text: None
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: None
}
thread 'test_vfs_works' panicked at 'Expected [VfsChange::RemoveFile { path: removed_path, .. }, VfsChange::AddFile {
text, path: added_path, .. }], got [RemoveFile { root: VfsRoot(1), file: VfsFile(4), path: "sub1/sub2/new.rs" }]', crates\ra_vfs\tests\vfs.rs:124:5
stack backtrace:
0: std::sys::windows::backtrace::set_frames
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\sys\windows\backtrace\mod.rs:104
1: std::sys::windows::backtrace::set_frames
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\sys\windows\backtrace\mod.rs:104
2: std::sys::windows::backtrace::set_frames
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\sys\windows\backtrace\mod.rs:104
3: std::sys_common::backtrace::print
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\sys_common\backtrace.rs:59
4: std::sys_common::backtrace::print
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\sys_common\backtrace.rs:59
5: std::panicking::default_hook
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\panicking.rs:221
6: std::panicking::rust_panic_with_hook
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\panicking.rs:491
7: std::panicking::continue_panic_fmt
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\panicking.rs:398
8: std::panicking::begin_panic_fmt
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libstd\panicking.rs:353
9: vfs::test_vfs_works
            at .\tests\vfs.rs:124
10: vfs::test_vfs_works::{{closure}}
            at .\tests\vfs.rs:30
11: core::ops::function::FnOnce::call_once<closure,()>
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libcore\ops\function.rs:238
12: test::run_test::{{closure}}
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libtest\lib.rs:1459
13: test::run_test::{{closure}}
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libtest\lib.rs:1459
14: test::run_test::{{closure}}
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libtest\lib.rs:1459
15: panic_unwind::__rust_maybe_catch_panic
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libpanic_unwind\lib.rs:102
16: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
17: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
18: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
19: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
20: std::thread::{{impl}}::spawn_unchecked::{{closure}}::{{closure}}
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\thread\mod.rs:477
21: std::thread::{{impl}}::spawn_unchecked::{{closure}}::{{closure}}
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\thread\mod.rs:477
22: std::thread::{{impl}}::spawn_unchecked::{{closure}}::{{closure}}
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\thread\mod.rs:477
23: panic_unwind::__rust_maybe_catch_panic
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\/src\libpanic_unwind\lib.rs:102
24: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
25: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
26: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
27: std::panicking::try
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\libstd\panicking.rs:289
28: alloc::boxed::{{impl}}::call_once
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\liballoc\boxed.rs:683
29: alloc::boxed::{{impl}}::call_once
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\liballoc\boxed.rs:683
30: alloc::boxed::{{impl}}::call_once
            at /rustc/9fda7c2237db910e41d6a712e9a2139b352e558b\src\liballoc\boxed.rs:683
31: BaseThreadInitThunk
32: RtlUserThreadStart


failures:
    test_vfs_works

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '-p ra_vfs --test vfs'
```

</details>

<details>
    <summary>Windows output with num_tasks 4</summary>

**NOTE** This fails only due to manually triggering assert!(false) to get the output at that point. Without the assertion, the test passes correctly.

```
running 1 test
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a/b\\baz.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Create(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1\\sub2"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1\\sub2\\new.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Rename(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1\\sub2\\new.rs",
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1\\sub2\\new1.rs"
    )
)
[crates\ra_vfs\src\io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "C:\\Users\\vvvv\\AppData\\Local\\Temp\\.tmppHSpnT\\a\\sub1\\sub2"
    )
)
test test_vfs_works ... FAILED

failures:

---- test_vfs_works stdout ----
[crates\ra_vfs\tests\vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        0
    ),
    files: [
        (
            "baz.rs",
            "nested hello"
        )
    ]
}
[crates\ra_vfs\tests\vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        1
    ),
    files: [
        (
            "bar.rs",
            "world"
        ),
        (
            "foo.rs",
            "hello"
        )
    ]
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "quux"
    )
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "corge"
    )
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: Some(
        "new hello"
    )
}
[crates\ra_vfs\tests\vfs.rs:120] "before rename" = "before rename"
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1",
    text: None
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: None
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new1.rs",
    text: Some(
        "new hello"
    )
}
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2",
    text: None
}
[crates\ra_vfs\tests\vfs.rs:133] "after rename" = "after rename"
thread 'test_vfs_works' panicked at 'assertion failed: false', crates\ra_vfs\tests\vfs.rs:135:5
```

</details>

<details>
    <summary>WSL output with num_tasks 2</summary>

**NOTE** This fails only due to manually triggering assert!(false) to get the output at that point. Without the assertion, the test passes correctly.
```
running 1 test
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "/tmp/.tmpr7WgK1/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "/tmp/.tmpr7WgK1/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "/tmp/.tmpr7WgK1/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "/tmp/.tmpr7WgK1/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Create(
        "/tmp/.tmpr7WgK1/a/sub1"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "/tmp/.tmpr7WgK1/a/sub1/sub2/new.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Rename(
        "/tmp/.tmpr7WgK1/a/sub1/sub2/new.rs",
        "/tmp/.tmpr7WgK1/a/sub1/sub2/new1.rs"
    )
)
test test_vfs_works ... FAILED

failures:

---- test_vfs_works stdout ----
[crates/ra_vfs/tests/vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        0
    ),
    files: [
        (
            "baz.rs",
            "nested hello"
        )
    ]
}
[crates/ra_vfs/tests/vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        1
    ),
    files: [
        (
            "bar.rs",
            "world"
        ),
        (
            "foo.rs",
            "hello"
        )
    ]
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "quux"
    )
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "corge"
    )
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: Some(
        "new hello"
    )
}
[crates/ra_vfs/tests/vfs.rs:120] "before rename" = "before rename"
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: None
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new1.rs",
    text: Some(
        "new hello"
    )
}
[crates/ra_vfs/tests/vfs.rs:133] "after rename" = "after rename"
thread 'test_vfs_works' panicked at 'assertion failed: false', crates/ra_vfs/tests/vfs.rs:135:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    test_vfs_works

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '-p ra_vfs --test vfs'
```

</details>

<details>
    <summary>WSL output with num_tasks 4</summary>

**NOTE** This should be expected, since there should be no more tasks to be processed.
```
running 1 test
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "/tmp/.tmpH7E7aN/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "/tmp/.tmpH7E7aN/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "/tmp/.tmpH7E7aN/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "/tmp/.tmpH7E7aN/a/b/baz.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Create(
        "/tmp/.tmpH7E7aN/a/sub1"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "/tmp/.tmpH7E7aN/a/sub1/sub2/new.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Rename(
        "/tmp/.tmpH7E7aN/a/sub1/sub2/new.rs",
        "/tmp/.tmpH7E7aN/a/sub1/sub2/new1.rs"
    )
)
test test_vfs_works ... FAILED

failures:

---- test_vfs_works stdout ----
[crates/ra_vfs/tests/vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        0
    ),
    files: [
        (
            "baz.rs",
            "nested hello"
        )
    ]
}
[crates/ra_vfs/tests/vfs.rs:11] &task = BulkLoadRoot {
    root: VfsRoot(
        1
    ),
    files: [
        (
            "bar.rs",
            "world"
        ),
        (
            "foo.rs",
            "hello"
        )
    ]
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "quux"
    )
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        0
    ),
    path: "baz.rs",
    text: Some(
        "corge"
    )
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: Some(
        "new hello"
    )
}
[crates/ra_vfs/tests/vfs.rs:120] "before rename" = "before rename"
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new.rs",
    text: None
}
[crates/ra_vfs/tests/vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2/new1.rs",
    text: Some(
        "new hello"
    )
}
thread 'test_vfs_works' panicked at 'called `Result::unwrap()` on an `Err` value: Timeout', src/libcore/result.rs:1009:5

```

</details>

**Comment by vipentti** [comment]
Upon closer inspection, it seems like `notify` on Windows reports some additional `NotifyWrite` and `Write` events for folders.

**NOTE** In the example outputs the paths have been manually trimmed of unnecessary path components and transformed to common format.

On Linux:

```
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Create(
        "a/sub1"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "a/sub1/sub2/new.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Rename(
        "a/sub1/sub2/new.rs",
        "a/sub1/sub2/new1.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "a/sub1/sub2/new1.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Remove(
        "a/sub1/sub2/new1.rs"
    )
)
```

Compared to Windows:

```
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Create(
        "a/sub1"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "a/sub1"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeWrite(
        "a/sub1/sub2"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "a/sub1/sub2/new.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "a/sub1"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Rename(
        "a/sub1/sub2/new.rs",
        "a/sub1/sub2/new1.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Write(
        "a/sub1/sub2"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    NoticeRemove(
        "a/sub1/sub2/new1.rs"
    )
)
[crates/ra_vfs/src/io.rs:165] ("convert_notify_event", &event) = (
    "convert_notify_event",
    Remove(
        "a/sub1/sub2/new1.rs"
    )
)
```

**Comment by vipentti** [comment]
Since we ignore `NotifyWrite` the problem seems to be the `Write` events surrounding the `Rename` event, these writes end up creating the extra tasks, even though these tasks end up doing nothing.

```
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1",
    text: None
}
// ...actual rename here ...
[crates\ra_vfs\tests\vfs.rs:11] &task = SingleFile {
    root: VfsRoot(
        1
    ),
    path: "sub1/sub2",
    text: None
}
```

**Comment by vipentti** [comment]
Possible fix in #828
